### ISSUE TITLE
TypedDict narrowing does not work with generic functions

### ISSUE THREAD

**Issue by roganov** [microsoft/pyright#7666]
**Describe the bug**
TypedDict `in` narrowing does not work with generic functions. In the code below, Pyright should not generate errors for the `go` function. 
This is admittedly pretty obscure, so please close if deemed unnecessary.

**Code or Screenshots**
```python
import typing


class My1(typing.TypedDict):
    a: int


class My2(typing.TypedDict):
    a: int
    b: int


Node1 = My1 | My2


class My3(typing.TypedDict):
    c: int


class My4(typing.TypedDict):
    c: int
    b: int


Node2 = My3 | My4


T = typing.TypeVar('T', Node1, Node2)


def go(n: T) -> None:
    if 'b' in n:
        n['b']  # error: Could not access item in TypedDict My1 and My3


def go2(n: Node1 | Node2) -> None:
    if 'b' in n:
        n['b']  # no error

```

For what it's worth, Mypy reports no errors as expected. (Although it requires decorating TypedDicts with `@typing.final`)

**VS Code extension or command-line**
command-line, 1.1.358

**Comment by erictraut** [comment]
This is included in pyright 1.1.359, which I just published.
